ini/hc32.ini

```diff
@@ -33,22 +33,14 @@ build_src_filter = ${common.default_src_filter} +<src/HAL/HC32> +<src/HAL/shared
 build_type       = release
 build_flags      =
   -D ARDUINO_ARCH_HC32
-  -D REDIRECT_PRINTF_TO_SERIAL              # Redirect core-provided printf to host serial
-  -D F_CPU=SYSTEM_CLOCK_FREQUENCIES.pclk1   # Override F_CPU to PCLK1, as marlin freaks out otherwise...
-  -D PLATFORM_M997_SUPPORT                  # Enable M997 command
+  -D PLATFORM_M997_SUPPORT  # Enable M997 command
+                            # note: ddl and arduino debug mode are 
+                            # automatically enabled with MARLIN_DEV_MODE
+  #-D __DEBUG               # force DDL debug mode
+  #-D __CORE_DEBUG          # force Arduino core debug mode
 
-  # DDL / Arduino Configuration
-  -D DISABLE_SERIAL_GLOBALS                 # Disable global Serial objects, we use our own
-  -D CORE_DISABLE_FAULT_HANDLER             # Disable arduino core fault handler (we use our own)
-
-  # DDL / Arduino Debug Options
-  #-D __DEBUG                               # DDL debug mode
-  #-D __CORE_DEBUG                          # Arduino core debug mode
-  -D PANIC_ENABLE                           # enable custom panic handlers (in MinSerial)
-  # options to reduce debug mode footprint (-16K; messages are less verbose)
-  -D __DEBUG_SHORT_FILENAMES                # Use short filenames in DDL debug output
-  -D __PANIC_SHORT_FILENAMES                # Use short filenames in core panic output
-  -D __OMIT_PANIC_MESSAGE                   # Omit panic messages in core panic output
+# hc32 app configuration file
+board_build.app_config = Marlin/src/HAL/HC32/app_config.h
```

Comment:
`board_build.app_config` is indeed unique to the HC32 arduino core, documentation [see here](https://github.com/shadow578/framework-arduino-hc32f46x/blob/main/docs/APP_CONFIG.md).

it's needed since some parts of the arduino core are configured using defines, and the `app_config.h` is included into all of the core's files by default.